enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,352 @@
+such as cost reduction, power efficiency, higher pod density and sustainability.
+
+The telco users have new use cases and requirements that especially related to resource management of the platform. 
+In order to address some of those needs,
+Kubelet CPU manager was created to allow users requesting for exclusive, isolated cpus.
+
+But there are additional use cases that require more granular control of CPU management.
+This enhancement aims to solve the following use case:
+When the CPU manager allocates an exclusive set of CPUs for a given workloads, the workload shall still have the option 
+to specify and access a set of shared CPUs, whether those shared CPUs are shared among containers in the 
+same pod or containers in different ones.
+
+By giving the users the flexibility to use exclusive CPUs only when it is an absolute necessity, we help them utilize
+their existing compute resources in an optimal manner.
+
+### User Stories
+
+* As a developer who's building DPDK application,
+I want to run my application in a container,
+giving full exclusive access to cpu cores for the busy-loops polling threads,
+but having the light-weight tasks like configuration processing and log printing not taking full cores,
+in order to improve the application density and make more efficient use of the CPUs.
+For that, my application container needs access to both shared and exclusive CPUs.
+
+  
+### Goals 
+* Increase pod density, power efficiency, cost reduction and sustainability by optimizing CPUs resource management. 
+* Provide a special-purpose at a runtime-level solution for containers in a pod to have the ability
+to request and subsequently be allocated both exclusive and shared CPUs.
+* Enable the telco-specific use case with a self-contained change,
+    minimizing the changes to the platform
+
+### Non-Goals
+* Introducing a generic mechanism in the platform that does involve Kubelet and pod spec changes.
+
+## Proposal
+
+Container processes are restricted
+to given [cpuset](https://man7.org/linux/man-pages/man7/cpuset.7.html) by their corresponding cgroup definition.
+Essentially, the kernel enforces the processes of container to run only at the CPU (core) ids specified under its
+cgroup.
+
+The mixed-cpu-node-plugin is a  runtime-agnostic, hybrid (composed of both [NRI-plugin](https://github.com/containerd/nri) and
+[device-plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)), 
+[mixed-cpu-node-plugin](https://github.com/openshift-kni/mixed-cpu-node-plugin) 
+that communicates with the container-runtime
+at various lifecycle stages of container e.g. CreateContainer, UpdateContainer, etc.,
+and modifies the container's cgroup to include a set of shared CPUs.
+
+Shared CPUs append to Kubelet [reservedSystemCpus](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#explicitly-reserved-cpu-list).
```

Comment:
if management and workload partitioning feature is enabled, the reservedSystemCpus were exclusive to management components (kubelet / crio) or pods that were labeled as management.  Are you intending these new 'shared' cpus to also support running kubelet/crio or pods labeled as 'management' or just pods that request the shared cpu?  What identifies which cpu is exclusive for management and/or exclusive for shared workload consumption in this set?

Reply:
Yes, I think an example with both enabled and how cpus are allocated will be useful here.

Reply:
> Are you intending these new 'shared' cpus to also support running kubelet/crio or pods labeled as 'management' or just pods that request the shared cpu?

When WP enabled there would be an absolute isolation between `reserved` and `shared` 
https://github.com/openshift/enhancements/pull/1396#discussion_r1199800642 
management components are pined to `reserved` cpus and `shared` are pined to pods which request shared cpus.

When WP disabled this is not the case,  management components/platform's housekeeping processes also runs on the `shared` (because `shared` are part of `systemReservedCPUs`) but pods which are ask for `shared` cpus cannot run on `reserved` because only `shared` cpus are exposed via cgroups as part of their cpuset.

>What identifies which cpu is exclusive for management and/or exclusive for shared workload consumption in this set?

In the performance profile:
`spec.cpu.reserved` = cpu is exclusive for management.
`spec.cpu.shared` = exclusive for shared workload consumption

I updated the EP (please check the last comment) accordingly and the Risk and Mitigation section as well.

> Yes, I think an example with both enabled and how cpus are allocated will be useful here.

I added a picture that describes cpu layout.